### PR TITLE
Restructure addon and example project

### PR DESCRIPTION
**Pull request by joelgomes1994**
## Overview
*This is a total rewrite, if approved it should be labeled as a v2 due to severe breaking changes.*

- Related to [this issue](https://github.com/Iakobs/godot-google-play-game-services-android-plugin/issues/3) and [this PR](https://github.com/Iakobs/godot-google-play-game-services-android-plugin/pull/4)
- I kept `assets/images` due to the usage in the [Asset Library](https://godotengine.org/asset-library/asset/1849) gallery

## Implementation
- [x] Rewrite GDScript plugin wrapper
    - [x] Join functionalities into a single autoload (to not clutter the global namespace with lots of singletons)
    - [x] Add support to [Events](https://developers.google.com/android/reference/com/google/android/gms/games/EventsClient) (for tracking and statistics)
    - [x] Add support to [Snapshots](https://developers.google.com/android/reference/com/google/android/gms/games/SnapshotsClient) (for saving and loading games)
- [x] Rewrite example project
    - [x] Structure project and assets in a separate folder
    - [x] Implement all new features from native plugin into the UI
    - [x] Cleanup and optimize assets from project folder
- [x] Rewrite README

## Example video
https://www.youtube.com/watch?v=RbEh7WLQ_d8

### CODE REVIEW

File: README.md
Comment (Iakobs) [inline README.md]: It is actually a Godot 4.2+ version. I haven't tried with versions between 4 and 4.2, but with 4.2 Godot introduced a change in how android plugins work, that's why I think (I didn't check) the other plugin won't work in versions lower to 4.2

Comment (joelgomes1994) [reply README.md]: Ok!

**Comment by joelgomes1994** [comment]
I was thinking that as we have only one autoload now, we could group signals and methods by using prefixes, to help on autocompletion. I did that on [this plugin](https://github.com/murikistudio/game-jolt-api).

These methods:
image

Would emit these signals:
image

What do you think about this idea?

**Comment by Iakobs** [comment]
> I was thinking that as we have only one autoload now, we could group signals and methods by using prefixes, to help on autocompletion. 
> What do you think about this idea?

I think it makes sense, it would be nice if you can add that change too but I feel like I'm abusing of your patience asking for so many things, thanks again for your help 🙏 

~By the way, this might come as a dumb question, but do you have permissions to approve the PR?~

edit: I just added a ruleset to prevent merge of PRs without 1 approval and sent you an invite to collaborate in both repos, this way you can merge the PR once it's approved!

**Review by Iakobs — DISMISSED** [review]
Let's do this! Once you merge the PR I will update the plugin to v2 and publish to the assets library. I will credit you as a collaborator as well. Thanks a lot! I will approve the other PR as well, and as soon as I can, I will add the events feature to the godot 4.2 plugin

**Review by Iakobs — DISMISSED** [review]

**Comment by joelgomes1994** [comment]
> Let's do this! Once you merge the PR I will update the plugin to v2 and publish to the assets library. I will credit you as a collaborator as well. Thanks a lot! I will approve the other PR as well, and as soon as I can, I will add the events feature to the godot 4.2 plugin

Alright! I just made the last update (I forgot to update the docs), now it's done.

Do you intend to make the 3.x and 4.x versions equal in features (method naming and other changes I made)? If so, I could rewrite and add the other features to the 4.x repository as well. If not, it's ok, as the 3.x version is the one I'm effectively using right now.

**Comment by Iakobs** [comment]
Hi! I want to put events in the godot 4.2 plugin, but I also want to try something different regarding the single autoload thing which I'm currently working on, so no worries about that plugin.

Thanks a lot!!

**Review by Iakobs — APPROVED** [review]

**Pull request merged**
